Documentation/network/node-ipam.rst

```diff
@@ -42,3 +42,101 @@ To enable the node IPAM on an existing installation, run:
      --reuse-values \\
      --set nodeIPAM.enabled=true
    kubectl -n kube-system rollout restart deployment/cilium-operator
+
+Gateway API and Ingress Controller Considerations
+-------------------------------------------------
+
+By Spec, Kubernetes requires at least one address in the Endpoints list if Pod Label selectors are not used.
+In order to overcome this limitation, Services created for Gateway API and Ingress Controller-managed resources,
+Cilium will inject a "virtual" Endpoint of ``192.192.192.192:9999``.
+
+Because all Cilium Agents are able to process Gateway API or Ingress traffic, it is possible for the Node IPAM LB
+to allow all Nodes in the Cluster to listen for incoming connections. Setting the ``loadBalancerClass`` to ``io.cilium/node``
+will add all Nodes to the Ingress IP list in the Load Balancer Service and accept traffic on the requested ports.
+
+Selecting Nodes to Listen for Gateway and Ingress Requests
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To restrict the Nodes that should listen for incoming traffic for Cilium Gateway and Ingress Services, you may
+add an annotation, ``io.cilium.nodeipam/match-labels``, to the Service. The value of the annotation is a
+`Label Selector <https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors>`__.
+
+.. note::
+   The node selector feature only works for Cilium-managed LoadBalancers for Gateways and Ingresses.
+   Applying this label has no effect on other LoadBalancer Services.
+
+
+You can test this Label Selector by using the ``kubectl get nodes`` command. For example, to exclude traffic to all
+Control Plane nodes, you may use this Label Selector: ``!node-role.kubernetes.io/control-plane``
+The Annotation would be configured as follows:
+
+.. parsed-literal::
+   io.cilium.nodeipam/match-labels="!node-role.kubernetes.io/control-plane"
+
+To verify the Nodes that would be returned, run the following command in your terminal:
+
+.. parsed-literal::
+   kubectl get nodes -l '!node-role.kubernetes.io/control-plane'
+
+   NAME          STATUS   ROLES    AGE     VERSION
+   kind-worker   Ready    <none>   6h32m   v1.29.1
+
+It is possible to combine multiple filters. For example, Nodes where the Label ``beehive.local/node`` value is
+``hive1`` or ``hive2``, the Label ``honey`` is set to ``yes``, and the Label ``bears`` does not exist would use the
+following filter Annotation:
+
+.. parsed-literal::
+   io.cilium.nodeipam/match-labels="beehive.local/node in (hive1,hive2),honey=yes,!bears"
+
+Configuring Resources for Node IPAM LB
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+**Ingress Controller**
```

Comment:
Could you put `Cilium Ingress Controller` to make it more obvious that we are talking specifically of the Cilium implementation here (same for GatewayAPI below)?